solutionbox/code_free_ml/transform_raw_data.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Flake8 cannot disable a warning for the file. Flake8 does not like beam code
+# and reports many 'W503 line break before binary operator' errors. So turn off
+# flake8 for this file.
+# flake8: noqa
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import datetime
+import json
+import logging
+import os
+import random
+import sys
+import apache_beam as beam
+from apache_beam.metrics import Metrics
+import six
+from tensorflow.python.lib.io import file_io
+from tensorflow_transform import coders
+from tensorflow_transform.beam import impl as tft
+from tensorflow_transform.beam import tft_beam_io
+from tensorflow_transform.tf_metadata import metadata_io
+
+
+img_error_count = Metrics.counter('main', 'ImgErrorCount')
+
+# Files
+SCHEMA_FILE = 'schema.json'
+FEATURES_FILE = 'features.json'
+
+TRANSFORMED_METADATA_DIR = 'transformed_metadata'
+RAW_METADATA_DIR = 'raw_metadata'
+TRANSFORM_FN_DIR = 'transform_fn'
+
+# Individual transforms
+TARGET_TRANSFORM = 'target'
+IMAGE_URL_TO_VEC_TRANSFORM = 'img_url_to_vec'
+
+
+def parse_arguments(argv):
+  """Parse command line arguments.
+  Args:
+    argv: list of command line arguments including program name.
+  Returns:
+    The parsed arguments as returned by argparse.ArgumentParser.
+  """
+  parser = argparse.ArgumentParser(
+      description='Runs Preprocessing on the Criteo model data.')
+
+  parser.add_argument(
+      '--project-id',
+      help='The project to which the job will be submitted.')
+  parser.add_argument(
+      '--cloud',
+      action='store_true',
+      help='Run preprocessing on the cloud.')
+  parser.add_argument(
+      '--job-name',
+      type=str,
+      help='Unique job name if running on the cloud.')
+
+  parser.add_argument(
+      '--csv-file-pattern',
+      required=False,
```

Comment:
I would avoid mentioning tf.example, since it is internal impl that users don't have to know. Maybe just "CSV data to transform".